Source/WebCore/Modules/url-pattern/URLPattern.cpp

```diff
@@ -324,4 +331,127 @@ ExceptionOr<void> URLPattern::compileAllComponents(ScriptExecutionContext& conte
     return { };
 }
 
+static inline void matchHelperAssignInputsFromURL(const URL& input, String& protocol, String& username, String& password, String& hostname, String& port, String& pathname, String& search, String& hash)
+{
+    protocol = input.protocol().toString();
+    username = input.user();
+    password = input.password();
+    hostname = !input.host().isNull() ? input.host().toString() : emptyString();
```

Comment:
Can we just do `hostname = input.host().toString()`? Do we really distinguish between empty string and null string in the existing implementation?